service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/PractitionerRoleMapper.java

```diff
@@ -47,8 +51,30 @@
         return roles;
     }
 
-    private CodeableConcept getCode(POCDMT000002UK01AssignedAuthor author) {
-        CE code = author.getCode();
+    public PractitionerRole mapResponsibleParty(POCDMT000002UK01ClinicalDocument1 clinicalDocument) {
```

Comment:
If result might not be there, a good practice is to return `Optional<PractitionerRole>` and then modify the code that uses this function as follows:
```
List<PractitionerRole> authorPractitionerRoles = practitionerRoleMapper.mapAuthorRoles(clinicalDocument.getAuthorArray());
practitionerRoleMapper.mapResponsibleParty(clinicalDocument)
    .ifPresent(practitionerRoleMapper::add);
Encounter encounter = encounterMapper.mapEncounter(clinicalDocument, practitionerRoles);
```